### PR TITLE
49628 handle BTSSS response exceptions

### PR DESCRIPTION
**Pull request by randomsync**
## Description of change

This change adds exception handling code for downstream BTSSS api returns a non-200 response.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#49628

## Things to know about this PR

- [x] rspec
- [ ] local integration testing

### CODE REVIEW

File: modules/check_in/spec/services/travel_claim/client_spec.rb [hunk 1]
```diff
@@ -123,11 +123,12 @@
         allow_any_instance_of(Faraday::Connection).to receive(:post).with(anything).and_raise(exception)
       end
 
-      it 'logs message and raises exception' do
+      it 'logs message and returns original error' do
         expect_any_instance_of(SentryLogging).to receive(:log_message_to_sentry)
-        expect do
-          subject.submit_claim(token: access_token, patient_icn: icn, appointment_date: appt_date)
-        end.to raise_exception(exception)
```
Comment (kanchanasuriya) [inline modules/check_in/spec/services/travel_claim/client_spec.rb:130]: Pls remove `exception` defined as Common::Exceptions::BackendServiceException in all the tests

Comment (randomsync) [reply modules/check_in/spec/services/travel_claim/client_spec.rb:130]: why? that's what's being raised by the connection to simulate the downstream error response.

Comment (kanchanasuriya) [reply modules/check_in/spec/services/travel_claim/client_spec.rb:130]: Got it, missed the usage in downstream error part

**Review by kanchanasuriya — APPROVED** [review]
LGTM

**Pull request merged**